File: app/src/main/java/com/link/fan/app/main/home/HomeAdapter.java
package com.link.fan.app.main.home;

import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import androidx.databinding.library.baseAdapters.BR;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.link.fan.R;
import com.link.fan.data.bean.MenuDetail;

import java.util.List;

public class HomeAdapter extends BaseQuickAdapter<MenuDetail, HomeAdapter.HomeViewHolder> {


    public HomeAdapter(int layoutResId, @Nullable List<MenuDetail> data) {
        super(layoutResId, data);
    }

    @Override
    protected void convert(HomeViewHolder helper, MenuDetail item) {
        ViewDataBinding binding = helper.getBinding();
        binding.setVariable(BR.menu, item);
        binding.setVariable(BR.clickListener, new View.OnClickListener() {

            @Override
            public void onClick(View v) {

            }
        });
        binding.executePendingBindings();
    }

    @Override
    protected View getItemView(int layoutResId, ViewGroup parent) {
        ViewDataBinding binding = DataBindingUtil.inflate(mLayoutInflater, layoutResId, parent, false);
        if (binding == null) {
            return super.getItemView(layoutResId, parent);
        }
        View view = binding.getRoot();
        view.setTag(R.id.BaseQuickAdapter_databinding_support, binding);
        return view;
    }


    public static class HomeViewHolder extends BaseViewHolder {

        public HomeViewHolder(View view) {
            super(view);
        }

        public ViewDataBinding getBinding() {
            return (ViewDataBinding) itemView.getTag(R.id.BaseQuickAdapter_databinding_support);
        }
    }

}

File: app/src/main/java/com/link/fan/app/main/home/HomeAdapter.kt
package com.link.fan.app.main.home

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.NavDirections
import androidx.navigation.findNavController
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.link.fan.data.bean.MenuDetail
import com.link.fan.databinding.ListItemHomeBinding

/**
 * copyright:TS
 * author:wujia
 * create:2019-11-06-11:10
 * email:wujia0916@thundersoft.com
 * description:
 */
class HomeAdapter : ListAdapter<MenuDetail, HomeAdapter.HomeHolder>(HomeDiffCallback()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HomeHolder {
        return HomeHolder(ListItemHomeBinding.inflate(LayoutInflater.from(parent.context), parent, false))
    }

    override fun onBindViewHolder(holder: HomeHolder, position: Int) {
        val menuResult = getItem(position)
        holder.bind(menuResult)
    }

    class HomeHolder(private val binding: ListItemHomeBinding) : RecyclerView.ViewHolder(binding.root) {

        init {
            binding.setClickListener {
                binding.menu?.let { menu ->
                    navigateToMenu(menu, it)
                }
            }
        }

        private fun navigateToMenu(menuDetail: MenuDetail, it: View) {
            //TODO
            val direction: NavDirections? = null
//            val direction = HomeViewPagerFragmentDirections.actionViewPagerFragmentToPlantDetailFragment(plant.plantId)
            it.findNavController().navigate(direction!!)
        }

        fun bind(item: MenuDetail) {
            binding.apply {
                menu = item
                executePendingBindings()
            }
        }

    }


}

private class HomeDiffCallback : DiffUtil.ItemCallback<MenuDetail>() {
    override fun areItemsTheSame(oldItem: MenuDetail, newItem: MenuDetail): Boolean {
        return oldItem.id == newItem.id
    }

    override fun areContentsTheSame(oldItem: MenuDetail, newItem: MenuDetail): Boolean {
        return oldItem == newItem
    }

}